### PR TITLE
feat: [SRM-13430]: adding complete validations for Health source framework

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
Validation Scenarios covered =>
**Validations should get triggered while :**

- Switching between metrics.
- Going to any other section outside custom metric.
- Clicking on Submit. (submission of health source)
- Showing the warning icon on the left panel whenever validation is not met.
- Should not allow switching of metrics if current form is not valid.
- Laying out the validation for  configurations form formik
- Laying out the validation for custom metric form formik.
- Don't show the left warning icon for the first load or render of the panel.
- Setting the query section to be touched or dirty whenever the validate method is called. (To ensure whenever we change panel or click outside without even touching the field , we should be able to see the validation.)

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/components/CommonMultiItemsSideNav/components/CommonSelectedAppsSideNav/components/GroupedSideNav/GroupedSideNav.utils.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import type { GroupedMetric } from './GroupedSideNav.types'
+
+export function showWarningIcon({
+  touched,
+  isValidInput,
+  selectedApp,
+  selectedItem
+}: {
+  touched: FormikTouched<unknown>
+  isValidInput: boolean | undefined
```
Comment (praneshg239) [inline src/modules/85-cv/components/CommonMultiItemsSideNav/components/CommonSelectedAppsSideNav/components/GroupedSideNav/GroupedSideNav.utils.ts:12]: use optional operator `isValidInput?: boolean`

Comment (harshilbits) [reply src/modules/85-cv/components/CommonMultiItemsSideNav/components/CommonSelectedAppsSideNav/components/GroupedSideNav/GroupedSideNav.utils.ts:12]: done.

**Comment by harshilbits** [comment]
retrigger jest

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.tsx [hunk 2]
```diff
@@ -88,25 +92,23 @@ export default function CustomMetricFormContainer(props: CustomMetricFormContain
...
-            clonedMappedMetricsData.delete('')
-          }
-          updateParentFormik(CommonHealthSourceContextFields.CustomMetricsMap, clonedMappedMetricsData)
-          updateParentFormik(CommonHealthSourceContextFields.SelectedMetric, selectedMetricName)
+          const updatedMappedMetricsData = getUpdatedMappedMetricsData(
+            mappedMetricsData,
+            selectedMetricName,
+            formValuesData
+          )
+          setFieldTouched(CustomMetricFormFieldNames.QUERY)
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.tsx:109]: why setting query field touched explicitly ?

Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.tsx:109]: To ensure whenever we change panel or click outside without even touching the field , we should be able to see the validation.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.utils.ts [hunk 3]
```diff
@@ -65,13 +69,40 @@ export function getAddMetricInitialValues(
...
+  selectedMetricName: string,
+  formValuesData: CommonCustomMetricFormikInterface
+): CommonHealthSourceConfigurations['customMetricsMap'] {
+  const updatedMappedMetricsData = cloneDeep(mappedMetricsData)
+  updatedMappedMetricsData.set(selectedMetricName, formValuesData)
+  cleanUpMappedMetrics(updatedMappedMetricsData)
+  return updatedMappedMetricsData
+}
+
+export function updateParentFormikWithLatestData(
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.utils.ts:101]: 👍

**Review by deepesh-ui — APPROVED** [review]

**Comment by praneshg239** [comment]
Please check whether the values are getting updated and validation is working properly while using `tab` button to change focus from one section to another section.

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**